examples/raytrain_nyctaxi.py

```diff
@@ -0,0 +1,176 @@
+    model.train()
+    train_loss, correct, data_size = 0, 0, 0
+    for batch_idx, (inputs, targets) in enumerate(dataset):
+        inputs, targets = inputs.to(device), targets.to(device)
+        # Compute prediction error
+        outputs = model(inputs)
+        loss = criterion(outputs, targets)
+        train_loss += loss.item()
+        correct += (outputs.argmax(1) == targets).type(torch.float).sum().item()
+        data_size += inputs.size(0)
+        # Backpropagation
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+    train_loss /= (batch_idx + 1)
+    train_acc = correct/data_size
+    return train_acc, train_loss
+
+def test_epoch(dataset, model, criterion, device):
+    model.eval()
+    # num_batches = len(dataset)
+    test_loss, correct, data_size = 0, 0, 0
+    with torch.no_grad():
+        for batch_idx, (inputs, targets) in enumerate(dataset):
+            inputs, targets = inputs.to(device), targets.to(device)
+            # Compute prediction error
+            outputs = model(inputs)
+            test_loss += criterion(outputs, targets).item()
+            correct += (outputs.argmax(1) == targets).type(torch.float).sum().item()
+            data_size += inputs.size(0)
+    test_loss /= (batch_idx + 1)
+    test_acc = correct/data_size
+    return test_acc, test_loss
+
+def train_func(config):
+    num_epochs = config["num_epochs"]
+    lr = config["lr"]
+    batch_size = config["batch_size"]
+    device = torch.device(f"cuda:{train.local_rank()}"
+                          if torch.cuda.is_available() else "cpu")
+    
+    # Then convert to torch datasets                
+    train_data_shard = get_dataset_shard("train")
+    train_dataset = train_data_shard.to_torch(feature_columns=features, label_column="fare_amount",
+                                               label_column_dtype=torch.float, feature_column_dtypes=feature_dtype, batch_size=batch_size)
+    test_data_shard = get_dataset_shard("test")
+    test_dataset = test_data_shard.to_torch(feature_columns=features, label_column="fare_amount",
+                                             label_column_dtype=torch.float, feature_column_dtypes=feature_dtype, batch_size=batch_size)
+    model = NYC_Model(len(features))
+    model = model.to(device)
+    model = train.torch.prepare_model(model)
+    
+    criterion = nn.SmoothL1Loss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=lr)
+    loss_results = []
+    for epoch in range(num_epochs):
+        train_acc, train_loss = train_epoch(train_dataset, model, criterion, optimizer, device)
+        test_acc, test_loss = test_epoch(test_dataset, model, criterion, device)
+        train.report(epoch = epoch, train_acc = train_acc, train_loss = train_loss)
+        train.report(epoch = epoch, test_acc=test_acc, test_loss=test_loss)
+        loss_results.append(test_loss)
+
+trainer = Trainer(backend="torch", num_workers=num_executors, use_gpu=False)
```

Comment:
Let num_workers be 2 to make it distributed. Claim num_cpus=4 in ray.init to pass CI.
@carsonwang Shall we include gpu support? We can't test it. This comes from a ray train example. If we support gpu, use_gpu should be a config, refer to the horovod example about how to set args.